src/main/frontend/src/components/aggregates/StateFields.svelte

```diff
@@ -9,14 +9,25 @@
   const stateFieldsTypes =  formatArrayForSelect(['int', 'double', 'String', 'float', 'short', 'byte', 'boolean', 'long', 'char']);
 
 	const addStateField = () => stateFields = stateFields.concat({ name: "", type: "" });
-	const deleteStateField = (index) => { stateFields.splice(index, 1); stateFields = stateFields; }
+  const deleteStateField = (index) => { stateFields.splice(index, 1); stateFields = stateFields; }
+  
+  const isFieldNameUnique = (value) => {
+    const result = stateFields.filter(field => field.name === value);
+		return result.length === 1 ? undefined : 'Name must be unique';
+  }
+
+  $: if (stateFields) {
+    stateFields.forEach(field => {
+      isFieldNameUnique(field.name);
```

Comment:
What are you trying to do here? The return value is not used.

Reply:
I will update this

Reply:
I solved this via using keyed each block